clients/src/main/java/org/apache/kafka/common/record/LegacyRecord.java

```diff
@@ -385,8 +385,7 @@ private static void write(ByteBuffer buffer,
                               ByteBuffer value,
                               CompressionType compressionType,
                               TimestampType timestampType) {
-        try {
-            DataOutputStream out = new DataOutputStream(new ByteBufferOutputStream(buffer));
+        try (DataOutputStream out = new DataOutputStream(new ByteBufferOutputStream(buffer))) {
```

Comment:
Note to reviewer(s):

This is a NoOp since `close()` method of `ByteBufferOutputStream` doesn't do anything. This change is for code hygiene.